task_processing/plugins/mesos/execution_framework.py

```diff
@@ -481,110 +267,95 @@ def create_new_docker_task(self, offer, task_config):
+                log.critical("Ignoring offer {offer_id} from blacklisted slave {slave_name}".format(
+                        offer_id=offer.id.value,
+                        slave_name=offer.slave_id.value
+                    )
+                )
+                driver.declineOffer(offer.id, self.offer_decline_filter)
+                continue
+
+            if not self.offer_matches_pool(offer):
+                log.info("Declining offer {id} because it is not for pool {pool}.".format(
+                        id=offer.id.value,
+                        pool=self.pool
                     )
-                    driver.declineOffer(offer.id)
-                    del offers[idx]
+                )
+                driver.declineOffer(offer.id, self.offer_decline_filter)
+                continue
 
-            if len(offers) == 0:
-                return
+            tasks_to_launch = self.get_tasks_to_launch(offer)
 
-            self.launch_tasks_for_offers(driver, offers)
+            if len(tasks_to_launch) == 0:
+                log.info("Declining offer {id} because it does not match our requirements.".format(id=offer.id.value))
+                driver.declineOffer(offer.id, self.offer_decline_filter)
+                continue
 
-    def statusUpdate(self, driver, update):
-        if self.queue:
-            self.queue.put(self.translator(update))
-        if self.frozen:
-            return
-
-        # update tasks
-        task_id = update.task_id.value
-        state = update.state
-        task_params = self.tasks_with_flags.setdefault(
-            task_id, MesosTaskParameters(health=None))
-        task_params.mesos_task_state = state
-
-        for task, params in list(self.tasks_with_flags.items()):
-            if params.marked_for_gc:
-                self.tasks_with_flags.pop(task)
-
-        if task_params.mesos_task_state is not TASK_STAGING:
-            if self.tasks_with_flags[task_id].staging_timer:
-                self.tasks_with_flags[task_id].staging_timer.cancel()
-                self.tasks_with_flags[task_id].staging_timer = None
-
-        # if task_params.mesos_task_state not in LIVE_TASK_STATES:
-            # task_params.marked_for_gc = True
-            # with self.constraint_state_lock:
-                # update_constraint_state(task_params.offer, self.constraints,
-                # self.constraint_state, step=-1)
+            log.info("Launching {number} new docker task(s) using offer {id} on slave {slave}".format(
+                    number=len(tasks_to_launch),
+                    id=offer.id.value,
+                    slave=offer.slave_id.value
+                )
+            )
+            driver.launchTasks(offer.id, tasks_to_launch)
 
-        driver.acknowledgeStatusUpdate(update)
+            for task in tasks_to_launch:
+                self.tasks_launched[task.task_id.value] = time.time()
+                self.task_to_slave_id_mappings[task.task_id.value] = task.slave_id.value
```

Comment:
i'd rather store a single compound value for each task instead of having to maintain multiple dictionaries